File: diancanba/diancanba-Restaurant/src/main/webapp/js/service/onecategoryService.js
//服务层
app.service('onecategoryService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../onecategory/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../onecategory/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../onecategory/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../onecategory/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../onecategory/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../onecategory/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../onecategory/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	//根据店铺id查询一级分类列表
	this.findByRestaurantId=function(){

		return $http.get('../onecategory/findOneCategoryList.do');
	}
});

File: diancanba/diancanba-Restaurant/src/main/java/com/jarvis/diancanba/restaurant/service/OnecategoryService.java
package com.jarvis.diancanba.restaurant.service;

import com.jarvis.diancanba.base.entity.PageResult;
import com.jarvis.diancanba.base.pojo.DcbOnecategory;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface OnecategoryService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<DcbOnecategory> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(DcbOnecategory onecategory);
	
	
	/**
	 * 修改
	 */
	public void update(DcbOnecategory onecategory);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public DcbOnecategory findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(DcbOnecategory onecategory, int pageNum, int pageSize);


	/**
	 *
	 * 根据id删除一节分类
	 */
	public void deleteById(Long id);


	public List<DcbOnecategory> findListByRestaurant(Long restaurantId);
}
